<a href="https://colab.research.google.com/github/katesheptak/DA_homework/blob/main/Workbook_of_hw_1_1_3_%D0%9E%D1%81%D0%BD%D0%BE%D0%B2%D0%B8_%D1%80%D0%BE%D0%B1%D0%BE%D1%82%D0%B8_%D0%B7_%D1%80%D1%8F%D0%B4%D0%BA%D0%B0%D0%BC%D0%B8_%D1%82%D0%B0_%D0%BC%D0%B0%D1%82%D0%B5%D0%BC%D0%B0%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D0%BE%D0%BF%D0%B5%D1%80%D0%B0%D1%86%D1%96%D1%97_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [2]:
a = 101
b = 210
c = 111
d = 402

result = ((a + b) * c) / d
print(result)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [3]:
a = 11283
b = 89
a%b

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [4]:
a = 11283
b = 89
print(a//b)

126


4. Запишіть у змінну `name` ваше імʼя.

In [10]:
#це моє Ім'я
a = 'Kateryna'
#або
b = "Kateryna"
print(a)
print(b)

Kateryna
Kateryna


5. Запишіть у змінну `surname` ваше прізвище.

In [13]:
#це моє Прізвище
a = 'Sheptak'
print(a)

Sheptak


6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [17]:
a = 'Kateryna'
b = 'Sheptak'
print(a + " " + b)

Kateryna Sheptak


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [20]:
a = 'Kateryna'
b = 'Sheptak'
print((a + " " + b).upper())

KATERYNA SHEPTAK


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [22]:
name_surname = "Kateryna Sheptak"
print(name_surname[2])

t


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [23]:
#перша літера
name_surname = "Kateryna Sheptak"
print(name_surname[0])

#остання літера
name_surname = "Kateryna Sheptak"
print(name_surname[-1])

K
k


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [30]:
surname = "Sheptak"
print(len(surname))

7


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [32]:
city = "Antananarivo"
count_a = city.lower().count('a')
print(count_a)

4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [33]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

# Перевірка чи починається рядок з "The"
starts_with_the = my_string.startswith("The")

# Перевірка чи закінчується рядок на "dog!"
ends_with_dog = my_string.endswith("dog!")

print("Починається з 'The':", starts_with_the)
print("Закінчується на 'dog!':", ends_with_dog)

Починається з 'The': True
Закінчується на 'dog!': False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [34]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

# Розділяємо рядок на список слів
words = my_string.split()

# З'єднуємо слова через нижнє підкреслювання
result = "_".join(words)

print(result)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`